### DISCUSSION TITLE
Is it possible to build jellyfin-web alone from source using Docker?

### DISCUSSION THREAD

**Discussion by Nicryc** [jellyfin/jellyfin-web#3841] [Q&A]
**See [update](#update)**

Hello,

I know I can run the Jellyfin server without the web instance using the `--nowebclient` option. However, is it possible to run the server on Windows using a binary and run the web client from source on Windows in a Docker container?

I've tried to build jellyfin-web from source using Docker but I didn't manage to do it. I cloned the jellyfin-web repo from GitHub and tried to build a Docker image using the `Dockerfile.portable` file:

```docker
docker build -f /mnt/c/Users/user/jellyfin-web-master/deployment/Dockerfile.portable .
```

The build succeed but the image is broken and isn't runnable. Here's the error message when running from Docker Desktop:

```
Error invoking remote method 'docker-run-container': Error: (HTTP code 400) unexpected - invalid reference format 
```

And from the WSL command line:

```
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/build.sh": stat /build.sh: no such file or directory: unknown.
ERRO[0001] error waiting for container: context canceled
```

It is very likely that I do that improperly, so how can I achieve that?

Thank you for your help.

## Update

OK I managed to build a working image using a custom Dockerfile:

```Dockerfile
FROM fedora:latest
EXPOSE 8080
COPY jellyfin-web-master /jellyfin-web
RUN dnf -y --refresh install nodejs npm && npm install -g npm 
CMD cd /jellyfin-web && npm install && npm start 
```

And running it like that:

```bash
docker run --name jellyfin-web-master -p 8080:8080 -it modified/jellyfin-web
```

Now the problem is when I launch my server with `--nowebclient` using the Windows binary and my web client in Docker I access the web UI at `localhost:8080`, I connect to my server at `localhost:8096` but I can't connect to an account. I always get an error even though I'm pretty sure I type my password correctly:

```
[15:02:54] [INF] [21] Jellyfin.Server.Implementations.Users.UserManager: Authentication request for user has been denied (IP: 127.0.0.1).
[15:02:54] [ERR] [21] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request: Invalid username or password entered. URL POST /Users/authenticatebyname.
```

**Comment by thornbill** [comment]
While you can certainly build jellyfin-web inside a docker container, running it with the development server vía `npm start` is not recommended for production use.

**Reply by Nicryc** [reply]
OK thank you for your reply. It's not a problem I'm just trying to build it inside a container to test some code.

Have you got an idea why it's throwing me these errors when I try to connect to an account?

**Comment by Chris-Codes-It** [comment]
Run both the server and UI as services on the same docker network; a good way would be to use a [docker compose file ](https://docs.docker.com/get-started/08_using_compose/)

So if the service name for the UI container was `jellyfin-web` and the service name for the server container was `jellyfin-server` you can connect on the url `jellyfin-server:8096`. You would still access the web front end via `localhost:8080` provided you've exposed the port, similiarly to how you've exposed 8080 in the DockerFile
